lib/lightning_web/components/viewers.ex

```diff
@@ -55,32 +68,117 @@ defmodule LightningWeb.Components.Viewers do
         </span>
       </div>
     <% else %>
-      <div
-        id={@id}
-        class={["flex grow", @class]}
-        phx-hook="LogViewer"
-        phx-update="ignore"
-        data-run-id={@run_id}
-        data-step-id={@selected_step_id}
-        data-loading-el={"#{@id}-nothing-yet"}
-        data-viewer-el={"#{@id}-viewer"}
-      >
-        <div class="relative grow">
-          <div
-            id={"#{@id}-nothing-yet"}
-            class="relative rounded-md p-12 text-center bg-slate-700 font-mono text-gray-200"
-          >
-            <.text_ping_loader>
-              Nothing yet
-            </.text_ping_loader>
+      <div class="relative flex grow">
+        <.log_level_filter
+          :if={@current_user}
+          id={"#{@id}-filter"}
+          levels={@selected_log_levels}
+        />
+
+        <div
+          id={@id}
+          class={["flex grow", @class]}
+          phx-hook="LogViewer"
+          phx-update="ignore"
+          data-run-id={@run_id}
+          data-step-id={@selected_step_id}
+          data-log-levels={Enum.join(@selected_log_levels, ",")}
+          data-loading-el={"#{@id}-nothing-yet"}
+          data-viewer-el={"#{@id}-viewer"}
+        >
+          <div class="relative grow">
+            <div
+              id={"#{@id}-nothing-yet"}
+              class="relative rounded-md p-12 text-center bg-slate-700 font-mono text-gray-200"
+            >
+              <.text_ping_loader>
+                Nothing yet
+              </.text_ping_loader>
+            </div>
+            <div id={"#{@id}-viewer"} class="hidden absolute inset-0 rounded-md">
+            </div>
           </div>
-          <div id={"#{@id}-viewer"} class="hidden absolute inset-0 rounded-md"></div>
         </div>
       </div>
     <% end %>
     """
   end
 
+  defp selected_log_levels(user) do
+    configured_levels = user.preferences["log_levels"] || []
+
+    if configured_levels == [] do
```

Comment:
Please notice it puts the [] to not use it. Would use  a `with` to nil, the default is good.

Reply:
I am imagining that `user.preferences["log_levels"]` will have either some _non empty_ list or nil (from a combox that has log levels or nil for users that has not set it yet)

Reply:
Combobox or checkboxes (: